WIP: Welcome: Standardize usage of 'Cell' ViewModels

Cliabhach opened the pull request:
This mostly just removes some duplicate code. There's also some better documentation of the various `All` fragments.

This PR is on hold until #297 and #299 are merged in. It will need some relatively-trivial updates after that point to account for moved and deleted files.

Cliabhach committed:
[Refactor] - Welcome: Use the same CellViewModels for both `home` and `all`

This removes a decent amount of code duplication.

Cliabhach committed:
[Refactor] - AllTours: remove obsolete color-setting code

Cliabhach committed:
[AllEvents] - Refactor: don't request an options menu, as we won't use it

Cliabhach committed:
[Documentation] - Welcome: add class-level docs to the 3 `All` fragments

caguilar187 commented on tours/src/main/kotlin/edu/artic/tours/AllToursViewModel.kt:
```diff
@@ -67,9 +67,9 @@ class AllToursViewModel @Inject constructor(
+ * ViewModel responsible for building each item in the tour summary list.
  */
-class AllToursCellViewModel(
+class TourCellViewModel(
```
let's move these into their own files now that they are being used in both places

Cliabhach replied:
Good idea.

caguilar187 commented on exhibitions/src/main/kotlin/edu/artic/exhibitions/AllExhibitionsFragment.kt:
```diff
@@ -50,6 +47,16 @@ class AllExhibitionsFragment : BaseViewModelFragment<AllExhibitionsViewModel>()
         val adapter = (recyclerView.adapter as AllExhibitionsAdapter)
+
+        /* Ensure search events go through ok. */
+        searchIcon
```
can you seperate this out into it's own pull request, i believe AllTours and AllEvents do the same thing so we should update those as well

Cliabhach replied:
Those other two already use this mechanism - AllExhibitions was the last hold-out. I'll create a PR for that and the layout pieces mentioned below.

Cliabhach replied:
It's up! Check out #297.

caguilar187 commented on exhibitions/src/main/res/layout/fragment_all_exhibitions.xml:
```diff
@@ -7,32 +7,38 @@
 
-    <android.support.design.widget.AppBarLayout
-        android:id="@+id/appBarLayout"
+    <android.support.v7.widget.Toolbar
```
similar to above lets create a layout we include for this and use in the three fragments. At least for that PR. After that we can use it in other places as well

Cliabhach replied:
This is done too! It's in commit e68b240b, merged in #299.

caguilar187 commented on welcome/src/main/kotlin/edu/artic/welcome/WelcomeEventsAdapter.kt:
```diff
@@ -14,8 +15,8 @@ import kotlinx.android.synthetic.main.welcome_event_cell_layout.view.*
 
-class WelcomeEventsAdapter : AutoHolderRecyclerViewAdapter<WelcomeEventCellViewModel>() {
-    override fun View.onBindView(item: WelcomeEventCellViewModel, position: Int) {
+class WelcomeEventsAdapter : AutoHolderRecyclerViewAdapter<EventCellViewModel>() {
```
since you are using the EventCellViewModel at this point and the cells are the same between both. Can we swap over to the EventsAdapter in the events package? The only difference is items and thats control in the WelcomeViewModel

Cliabhach replied:
That change will need to be covered as a separate part of #298, so I'll close this particular comment as resolved within the scope of #291.

Cliabhach committed:
[ContentListing] - Refactor: rename one of the AllToursItemDecoration.kt files

I don't know why this wasn't caught earlier? Must've not been
reviewed properly.

Cliabhach committed:
[ContentListing] - Refactor: move some files to a new, dedicated package

Cliabhach committed:
[ContentListing] - Refactor: remove two minor rounding operations in ATID

Cliabhach committed:
[ContentListing] - Refactor: convert two `position` vars into vals

With this, all of the local variables in the `decoration` package are
immutable.

Cliabhach committed:
[ContentListing] - Documentation: actually attribute those files to the author

Cliabhach committed:
[ContentListing] - Refactor: inline safe assumptions about the decoration code

Cliabhach committed:
[Refactor] - Use prebuilt `Rect::setEmpty` instead of `Rect::set(0,0,0,0)`

Cliabhach committed:
[ContentListing] - Create new base class for the three decorations

Cliabhach committed:
[ContentListing] - Use that new class to late-initialize dimensions

Cliabhach committed:
[ContentListing] - Remove now-unneccessary constructor parameter
